jobs/cluster-6/templates/bin/post-deploy

```diff
@@ -1,19 +1,46 @@
 #!/bin/bash
+
+function log() {
+  echo "[$(date)] $*"
+}
+
 <% if !spec.bootstrap %>log "this node is not the bootstrap node; nothing to do."
 exit 0
 <% else %>set -eu
 exec >&2
 
+<% if p('redis.tls.enabled') -%>
+# Install and setup stunnel.
+apt install -y redis-tools stunnel4
+
 function redis() {
   local ip=$1 ; shift
+
+  # We need to setup stunnel for TLS connections.
+  cat > /etc/stunnel/stunnel.conf << EOF
+[redis-cli]
+  client = yes
+  accept = 127.0.0.1:8000
+  connect = $ip:<%= p('redis.tls.dual-mode') ? 16379 : 6379 %>
+EOF
+
+  # Restart stunnel to apply changes.
+  kill `cat /var/run/stunnel4.pid` || true
```

Comment:
Never use backticks for sub-processes in shell scripting, always use `$()` for subshells. 
Backticks have error and edge cases that can surprise when least expected whereas the `$()` always work as intended.

Reply:
Thanks! I copied this block from the `stunnel` official docs.